### PR TITLE
[DOC-252] Added pg_locks and moved some pages from query tuning to observability 

### PR DESCRIPTION
**Pull request by aishwarya24**
Slack request - https://yugabyte.slack.com/archives/CDQB8MEAU/p1707938328365129
https://deploy-preview-21231--infallible-bardeen-164bc9.netlify.app/stable/explore/

Pg_locks-https://deploy-preview-21231--infallible-bardeen-164bc9.netlify.app/stable/explore/observability/pg-locks/

### CODE REVIEW

File: docs/content/stable/explore/observability/pg-locks.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+### waitend
+
+The waitend field contains the timestamp at which the lock was acquired by a transaction (PostgtreSQL does not record this timestamp currently.)
+
+### ybdetails
+
+The ybdetails field is a JSONB type that encapsulates additional information about each lock, specific to YugabyteDB's distributed architecture, and includes the following attributes:
+
+- `node`: The YB-TServer UUID of the node hosting the PostgreSQL backend that is holding the lock. It helps in identifying the specific node in the YugabyteDB cluster where the lock is being held, facilitating easier troubleshooting and monitoring of lock distribution across nodes.
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:54]:
```suggestion
- `node`: The YB-TServer UUID of the node hosting the PostgreSQL backend that is holding the lock. Identifies the specific node in the YugabyteDB cluster where the lock is being held, which allows you to troubleshoot and monitor lock distribution across nodes.
```

File: docs/content/stable/explore/observability/pg-locks.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+- `transactionid`: The UUID of the YugabyteDB transaction ID owning this lock. This field is NULL for single shard or fastpath operations, offering visibility into the transactional context of the lock. It aids in tracking and managing transactions across the distributed database.
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:55]:
```suggestion
- `transactionid`: The UUID of the YugabyteDB transaction ID owning this lock. This field is NULL for single shard or fastpath operations, offering visibility into the transactional context of the lock. Use this field to track and manage transactions across the distributed database.
```

File: docs/content/stable/explore/observability/pg-locks.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+- `subtransaction_id`: The ID of the subtransaction in which the lock was acquired.
+- `is_explicit`: True when the lock was acquired explicitly, such as through FOR UPDATE, FOR NO KEY UPDATE, FOR SHARE, FOR KEY SHARE, and so on. This attribute helps distinguish between locks acquired automatically by the database system and those requested explicitly by you, aiding in lock analysis and optimization efforts.
+- `tablet_id`: The ID of the tablet containing this lock. In YugabyteDB, data is sharded into tablets, and this ID helps identify the specific shard where the lock exists, crucial for diagnosing sharding-related lock contention issues.
+- `blocked_by`: A list of transactions blocking the acquisition of this lock. This field is helpful in identifying deadlock scenarios, and determining which transaction is blocking other operations from moving forward.
+- `keyrangedetails`: Provides details about what keys the lock is held, including the following:
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:60]: "what keys the lock is held" this fragment doesn't quite parse

File: docs/content/stable/explore/observability/pg-locks.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+  - `cols`: A list of column values from the PRIMARY KEY of the table, offering insight into the exact row(s), or key range locked.
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:62]:
```suggestion
  - `cols`: A list of column values from the PRIMARY KEY of the table, offering insight into the exact row(s) or key range that is locked.
```

File: docs/content/stable/explore/observability/pg-locks.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  - `attnum`: The PostgreSQL attribute number indicating if the lock is a column-level lock, linking the lock to the specific table column.
+  - `column_id`: The column ID in DocDB if the lock is a column-level lock, further aligning lock information with YugabyteDB's internal document-oriented architecture.
+  - `multiple_rows_locked`: Indicates when the lock is held on more than one entry in DocDB, helping to understand the scope of the lock in the database's document model.
+
+## Configurable parameters for lock management
+
+YugabyteDB offers several GUC parameters to customize how locks are queried and displayed, so you can tailor the lock information as needed. `yb_locks_min_txn_age` and `yb_locks_max_transactions`, control the filtering and limitation of transactions in lock queries.
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:69]:
```suggestion
YugabyteDB offers several GUC parameters to customize how locks are queried and displayed, so you can tailor the lock information as needed. `yb_locks_min_txn_age` and `yb_locks_max_transactions` control the filtering and limitation of transactions in lock queries.
```

File: docs/content/stable/explore/observability/pg-locks.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+Example: You can use the following command to adjust the minimum transaction age to 5 seconds, thereby excluding more recent transactions from lock queries:
+
+```sh
+SET session yb_locks_min_txn_age = 5000;
+```
+
+### yb_locks_max_transactions
+
+The `yb_locks_max_transactions` parameter sets the maximum number of transactions for which lock information is displayed when you query the pg_locks view. It allows to limit the output to the most relevant transactions, which is particularly beneficial in environments with high levels of concurrency and transactional activity. By controlling the volume of information returned, this parameter helps in managing the analysis of lock contention more effectively.
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:85]:
```suggestion
The `yb_locks_max_transactions` parameter sets the maximum number of transactions for which lock information is displayed when you query the pg_locks view. You can limit the output to the most relevant transactions, which is particularly beneficial in environments with high levels of concurrency and transactional activity. By controlling the volume of information returned, this parameter helps in managing the analysis of lock contention more effectively.
```

File: docs/content/stable/explore/observability/pg-locks.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+  - `column_id`: The column ID in DocDB if the lock is a column-level lock, further aligning lock information with YugabyteDB's internal document-oriented architecture.
+  - `multiple_rows_locked`: Indicates when the lock is held on more than one entry in DocDB, helping to understand the scope of the lock in the database's document model.
+
+## Configurable parameters for lock management
+
+YugabyteDB offers several GUC parameters to customize how locks are queried and displayed, so you can tailor the lock information as needed. `yb_locks_min_txn_age` and `yb_locks_max_transactions`, control the filtering and limitation of transactions in lock queries.
+
+### yb_locks_min_txn_age
+
+The `yb_locks_min_txn_age` parameter specifies the minimum age of a transaction (in seconds) before its locks gets included in the results returned from querying the `pg_locks` view. By setting this parameter, you can focus on older transactions that may be more relevant to performance tuning or deadlock resolution efforts. Transactions that are started more recently than the specified duration is not shown, helping to reduce clutter and focus on potentially problematic transactions.
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:73]:
```suggestion
The `yb_locks_min_txn_age` parameter specifies the minimum age of a transaction (in seconds) before its locks are included in the results returned from querying the `pg_locks` view. By setting this parameter, you can focus on older transactions that may be more relevant to performance tuning or deadlock resolution efforts. Transactions that are started more recently than the specified duration are not shown, helping to reduce clutter and focus on potentially problematic transactions.
```

File: docs/content/stable/explore/observability/pg-locks.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+Default: 1 second
+
+Example: You can use the following command to adjust the minimum transaction age to 5 seconds, thereby excluding more recent transactions from lock queries:
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:77]:
```suggestion
Example: Enter the following command to change the minimum transaction age to 5 seconds:
```

File: docs/content/stable/explore/observability/pg-locks.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+SET session yb_locks_min_txn_age = 5000;
+```
+
+### yb_locks_max_transactions
+
+The `yb_locks_max_transactions` parameter sets the maximum number of transactions for which lock information is displayed when you query the pg_locks view. It allows to limit the output to the most relevant transactions, which is particularly beneficial in environments with high levels of concurrency and transactional activity. By controlling the volume of information returned, this parameter helps in managing the analysis of lock contention more effectively.
+
+Default: 16
+
+Example: You can use the following command to change the maximum number of transactions to display to 10:
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:89]:
```suggestion
Example: Enter the following command to change the maximum number of transactions to display to 10:
```

File: docs/content/stable/explore/observability/pg-locks.md [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+```sh
+SET session yb_locks_max_transactions = 10;
+```
+
+## Examples
+
+Following are some examples of how to use the pg_locks view in YugabyteDB.
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:97]:
```suggestion
The following examples show how you can use the pg_locks view in YugabyteDB:
```

File: docs/content/stable/explore/observability/pg-locks.md [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+{{% explore-setup-single %}}
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:99]: move this above the paragraph

File: docs/content/stable/explore/query-1-performance/_index.md [hunk 12]
```diff
@@ -9,7 +9,7 @@ menu:
   stable:
     identifier: query-tuning
     parent: explore
-    weight: 280
+    weight: 310
```
Comment (ddhodge) [inline docs/content/stable/explore/query-1-performance/_index.md:12]: maybe put Query tuning right after observability; then security

File: docs/content/stable/explore/observability/pg-locks.md [hunk 13]
```diff
@@ -0,0 +1,138 @@
+---
+title: Get lock information insights with pg_locks
```
Comment (ddhodge) [inline docs/content/stable/explore/observability/pg-locks.md:2]: need to think of a better title

File: docs/content/stable/reference/configuration/yb-tserver.md [hunk 14]
```diff
@@ -1204,6 +1204,20 @@ Default: `false`
 Use of this flag can potentially result in expiration of live data. Use at your discretion.
 {{< /warning >}}
 
+## Locking flags
+
+##### yb_locks_min_txn_age
```
Comment (ddhodge) [inline docs/content/stable/reference/configuration/yb-tserver.md:1209]: these are flags as well as config parameters?

File: docs/content/stable/explore/observability/pg-locks.md [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+menu:
+  stable:
+    identifier: pg-locks
+    parent: explore-observability
+    weight: 410
+type: docs
+---
+
+YugabyteDB supports the PostgreSQL [pg_locks](https://www.postgresql.org/docs/current/view-pg-locks.html) system view that provides information about the locks held and requested by the current active transactions in a database. YugabyteDB enhances this view with two new fields `waitend` and `ybdetails` offering insights into lock information specific to YugabyteDB's distributed architecture. The `pg_locks` view is tailored to YugabyteDB's lock handling mechanisms, providing a comprehensive overview of database lock states.
+
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/stable/explore/observability/pg-locks.md:15]: examples are at the end of the page.

File: docs/content/stable/explore/query-1-performance/_index.md [hunk 16]
```diff
@@ -9,7 +9,7 @@ menu:
   stable:
     identifier: query-tuning
     parent: explore
-    weight: 280
+    weight: 300
 type: indexpage
```
Comment (ddhodge) [inline docs/content/stable/explore/query-1-performance/_index.md:13]: Child pages in this section now have broken links to pages that moved. Do a global search for the pages and fix

File: docs/content/stable/reference/configuration/yb-tserver.md [hunk 17]
```diff
@@ -1265,6 +1265,20 @@ You can modify these parameters in the following ways:
     SET LOCAL temp_file_limit=-1;
     ```
 
+- To specify the minimum age of a transaction (in seconds) before its locks gets included in the results returned from querying the [pg_locks](../../../explore/observability/pg-locks/) view, use [yb_locks_min_txn_age](../../../explore/observability/pg-locks/#yb-locks-min-txn-age):
```
Comment (ddhodge) [inline docs/content/stable/reference/configuration/yb-tserver.md:1268]:
```suggestion
- To specify the minimum age of a transaction (in seconds) before its locks are included in the results returned from querying the [pg_locks](../../../explore/observability/pg-locks/) view, use [yb_locks_min_txn_age](../../../explore/observability/pg-locks/#yb-locks-min-txn-age):
```

**Review by sushantrmishra — APPROVED** [review]

File: docs/content/stable/explore/query-1-performance/_index.md [hunk 18]
```diff
@@ -9,10 +9,97 @@ menu:
...
+
+The `pg_stat_statements` extension module is installed by default, but must be enabled for a database before you can query the `pg_stat_statements` view.
+
+```sql
+CREATE EXTENSION if not exists pg_stat_statements;
+```
+
+To get the output of `pg_stat_statements` in JSON format, visit `https://<yb-tserver-ip>:13000/statements` in your web browser, where `<yb-tserver-ip>` is the IP address of any YB-TServer node in your cluster.
+
+For more information, refer to [Get query statistics using `pg_stat_statements`](../pg-stat-statements).
```
Comment (ddhodge) [inline docs/content/stable/explore/query-1-performance/_index.md:30]:
```suggestion
For more information, refer to [Get query statistics using `pg_stat_statements`](./pg-stat-statements/).
```

File: docs/content/preview/explore/query-1-performance/pg-hint-plan.md [hunk 19]
```diff
@@ -602,6 +602,6 @@ This example inserts queries into the `hint_plan.hints` table, with placeholders
 ## Learn more
```
Comment (ddhodge) [inline docs/content/preview/explore/query-1-performance/pg-hint-plan.md:602]: fix link line 47

File: docs/content/preview/explore/observability/_index.md [hunk 20]
```diff
@@ -32,36 +32,44 @@ YugabyteDB exports various [metrics](../../launch-and-manage/monitor-and-alert/m
...
 
-{{<note>}}
+{{<tip>}}
 Both [YugabyteDB Anywhere](../../yugabyte-platform/alerts-monitoring/anywhere-metrics/) and [YugabyteDB Managed](../../yugabyte-cloud/cloud-monitor/overview/) come with a full suite of visualizations to help you monitor your cluster and troubleshoot issues.
-{{</note>}}
+{{</tip>}}
 
 ## Logging
 
 Logs from different services, such as the [YB-TServer](../../troubleshoot/nodes/check-logs/#yb-tserver-logs) and [YB-Master](../../troubleshoot/nodes/check-logs/#yb-master-logs) provide a historical record of what has happened and can be very helpful in debugging and troubleshooting. These logs are rotated regularly, based on their size as configured. See [Logs management](../../troubleshoot/nodes/check-logs#logs-management).
```
Comment (ddhodge) [inline docs/content/preview/explore/observability/_index.md:60]: these links should be preview as well then I guess?

**Pull request merged**